### PR TITLE
Python 3

### PR DESCRIPTION
**Pull request by LePetitTim**
Passage en python 3.

### CODE REVIEW

**Comment by leplatrem** [comment]
Thanks for this!

Apparently Mapnik 2.2 cannot be installed on Trusty, you may have to change Travis to install `ppa:mapnik/nightly-2.3`

**Comment by LePetitTim** [comment]
The mapnik doesn't work for now.

**Comment by leplatrem** [comment]
> The mapnik doesn't work for now.

It makes me wonder if the mapnik rendering is a popular feature :)

**Comment by LePetitTim** [comment]
There is a problem with travis ... 
I don't know why.

**Comment by fykidwai** [comment]
When will this be merged?

File: landez/tests.py [hunk 1]
```diff
@@ -22,12 +22,13 @@ def test_format(self):
         # Format from WMS options
         mb = TilesManager(wms_server='dumb', wms_layers=['dumber'],
                           wms_options={'format': 'image/jpeg'})
+
         self.assertEqual(mb.tile_format, 'image/jpeg')
-        self.assertEqual(mb.cache.extension, '.jpeg')
+        self.assertTrue(mb.cache.extension, '.jpeg')
```
Comment (submarcos) [inline landez/tests.py]: pourquoi True ?

File: landez/tests.py [hunk 2]
```diff
@@ -22,12 +22,13 @@ def test_format(self):
...
         # Format from URL extension
         mb = TilesManager(tiles_url='http://tileserver/{z}/{x}/{y}.jpg')
         self.assertEqual(mb.tile_format, 'image/jpeg')
-        self.assertEqual(mb.cache.extension, '.jpeg')
+        self.assertTrue(mb.cache.extension, '.jpeg')
```
Comment (submarcos) [inline landez/tests.py]: pourquoi True ?

File: landez/tests.py [hunk 3]
```diff
@@ -103,8 +104,14 @@ class TestMBTilesBuilder(unittest.TestCase):
     def tearDown(self):
         try:
             shutil.rmtree(self.temp_cache)
+        except OSError:
```
Comment (submarcos) [inline landez/tests.py]: if os.path.exists ?

File: landez/tests.py [hunk 4]
```diff
@@ -103,8 +104,14 @@ class TestMBTilesBuilder(unittest.TestCase):
...
+            pass
+        try:
             shutil.rmtree(self.temp_dir)
-            os.remove('foo.mbtiles')
+        except OSError:
```
Comment (submarcos) [inline landez/tests.py]: if os.path.exists ?

File: landez/tiles.py [hunk 5]
```diff
@@ -115,7 +115,7 @@ def __init__(self, **kwargs):
         # Tile files extensions
         self._tile_extension = mimetypes.guess_extension(self.tile_format, strict=False)
         assert self._tile_extension, _("Unknown format %s") % self.tile_format
-        if self._tile_extension == '.jpe':
+        if self._tile_extension in ('.jpe', '.jpg'):
```
Comment (submarcos) [inline landez/tiles.py]: .jpg / .jpeg ? (au lieu de .jpe)

File: setup.py [hunk 6]
```diff
@@ -28,7 +28,7 @@
     include_package_data=True,
     zip_safe=False,
     keywords=['MBTiles', 'Mapnik'],
-    classifiers=['Programming Language :: Python :: 2.6',
+    classifiers=['Programming Language :: Python :: 3.5',
```
Comment (submarcos) [inline setup.py]: il faut ajouter python 3.6 mais garder le 2.x (au passage mettre 2.7)

File: landez/sources.py [hunk 7]
```diff
@@ -6,11 +6,16 @@
...
-import urllib
 import requests
-from urlparse import urlparse
+try:
+    from urllib.parse import urlparse, urlencode
+    from urllib.request import urlopen, Request
+except ImportError:
+    from urlparse import urlparse
+    from urllib import urlencode
+    from urllib2 import urlopen, Request
```
Comment (submarcos) [inline landez/sources.py]: c'est pas obligatoire mais la bibliotheque six est faite pour çà

**Pull request merged**